# Problema de optimizacion para una carpinteria


Se necesita optimizar la funcion de utilidad de una carpinteria la cual tiene recursos limitados.
Para la elaboracion de los muebles se debe tomar en cuenta dos tipos de de trabajo: Carpinteria y y acabado.
Cada restriccion va a variar de acuerdo al mueble producido (sillas o mesas).

x1 = variable de sillas
x2 = variable de mesas

Funcion de utilidad a maximizar: 20x1+30x2
Restriccion de horas de acabado: x1+2x2 <=100
Restriccion de horas de carpinteria: 2x1+x2 <=100

In [69]:
from pulp import *
from matplotlib import*

prob = LpProblem("Utilidad",LpMaximize) #creacion del problema de maximizacion en prog. lineal.

x1 = LpVariable("x1", lowBound=0, cat ='Integer') # creacion de la variable mueble silla X1 >=0 (entero)
x2 = LpVariable("x2", lowBound=0, cat ='Integer') #creacion de la variable mueble mesa x2 >=0 (entero)



prob += 20*x1 + 30*x2 # Funcion objetivo (Funcion de utilidad) 
prob += 1*x1 + 2*x2 <= 100 # Restriccion de horas de acabado disponibles
prob += 2*x1 + 1*x2  <= 100 # Restriccion de horas de carpinteria disponibles
prob


Utilidad:
MAXIMIZE
20*x1 + 30*x2 + 0
SUBJECT TO
_C1: x1 + 2 x2 <= 100

_C2: 2 x1 + x2 <= 100

VARIABLES
0 <= x1 Integer
0 <= x2 Integer

In [70]:
status = prob.solve()
LpStatus[status]


'Optimal'

In [71]:
value(x1), value(x2), value(prob.objective) # ver los parametros de la solucion, 

(32.0, 34.0, 1660.0)

Se configuro al programa para que de resultados enteros debido a que las sillas y mesas no pueden ser variables continuas.
Los resutlados optimos son: 32 silas, 34 mesas y como utilidad maxima se tiene 1660 unidades monetarias.